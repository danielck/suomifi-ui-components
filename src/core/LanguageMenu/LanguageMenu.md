```js
import {
  LanguageMenu,
  LanguageMenuItem
} from 'suomifi-ui-components';

const languages = [
  { text: 'Suomeksi (FI)', lang: 'fi' },
  { text: 'PÃ¥ svenska (SV)', lang: 'sv' },
  { text: 'In English (EN)', lang: 'en' }
];

const [selectedLanguage, setSelectedLanguage] = React.useState(
  languages[2]
);

<LanguageMenu
  buttonText={selectedLanguage.text}
  aria-label={`Change language, selected language: ${selectedLanguage.text}`}
>
  {languages.map((item, index) => (
    <LanguageMenuItem
      onSelect={() => setSelectedLanguage(languages[index])}
      lang={languages[index].lang}
      key={languages[index].lang}
      selected={selectedLanguage.lang === item.lang}
    >
      {languages[index].text}
    </LanguageMenuItem>
  ))}
</LanguageMenu>;
```
