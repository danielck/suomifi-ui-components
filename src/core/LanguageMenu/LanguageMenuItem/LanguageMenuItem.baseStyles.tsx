import { css } from 'styled-components';
import { SuomifiTheme } from '../../theme';
import { element } from '../../theme/reset';

export const baseStyles = (theme: SuomifiTheme) => css`
  /* stylelint-disable no-descending-specificity */
  &.fi-language-menu-item {
    ${element(theme)}
    margin: ${theme.spacing.xxs} 0;

    .fi-link--router {
      ${theme.typography.actionElementInnerText}
      word-break: break-word;
      color: ${theme.colors.blackBase};
      display: inline-block;
      width: 100%;
      padding: 0 ${theme.spacing.m} 0 ${theme.spacing.xxs};
      border-left: 4px solid transparent;

      &:hover {
        text-decoration: none;
      }
    }

    &--isHighlighted {
      .fi-link--router {
        border-left-color: ${theme.colors.highlightBase};
        position: relative;

        &:after {
          content: '';

          @media (forced-colors: active) {
            position: absolute;
            left: 1px;
            right: 1px;
            top: 1px;
            bottom: 1px;
            border: solid 3px Highlight;
          }
        }
      }
    }

    &--selected {
      .fi-link--router {
        font-weight: bold;
        border-left-color: ${theme.colors.highlightBase};
      }
    }
  }
`;
