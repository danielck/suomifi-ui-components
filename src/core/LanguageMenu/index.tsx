export { LanguageMenu, LanguageMenuProps } from './LanguageMenu';
export {
  LanguageMenuItem,
  LanguageMenuItemProps,
} from './LanguageMenuItem/LanguageMenuItem';
