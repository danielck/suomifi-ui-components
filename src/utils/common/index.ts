export {
  windowAvailable,
  useEnhancedEffect,
  getOwnerDocument,
  escapeStringRegexp,
  forkRefs,
  autocompleteTimeString,
} from './common';
