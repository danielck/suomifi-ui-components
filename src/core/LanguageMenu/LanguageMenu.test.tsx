import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import { axeTest } from '../../utils/test';
import { LanguageMenu, LanguageMenuProps } from './LanguageMenu';
import { LanguageMenuItem } from './LanguageMenuItem/LanguageMenuItem';

const doNothing = () => ({});

const languageMenuProps: LanguageMenuProps = {
  buttonText: 'In English (EN)',
  className: 'lm-test',
  id: 'test-id',
  wrapperProps: {
    id: 'wrapper-id',
    'data-testid': 'language-menu-test-id',
  },
};

const TestLanguageMenu = (props: LanguageMenuProps) => (
  <LanguageMenu {...props}>
    <LanguageMenuItem onClick={() => doNothing()} lang="fi">
      Suomeksi (FI)
    </LanguageMenuItem>
    <LanguageMenuItem href="/sv" lang="sv">
      PÃ¥ svenska (SV)
    </LanguageMenuItem>
    <LanguageMenuItem href="/en" lang="en" selected>
      In English (EN)
    </LanguageMenuItem>
  </LanguageMenu>
);

describe('Basic LanguageMenu', () => {
  const BasicLanguageMenu = TestLanguageMenu(languageMenuProps);

  it('should have provided ids', async () => {
    const { findByRole, findByTestId } = render(BasicLanguageMenu);
    const wrapperDiv = await findByTestId('language-menu-test-id');
    expect(wrapperDiv).toBeTruthy();
    expect(wrapperDiv).toHaveAttribute('id', 'wrapper-id');
    const button = await findByRole('button');
    expect(button).toHaveAttribute('id', 'test-id');
  });

  it('should have button text', async () => {
    const { findByRole } = render(BasicLanguageMenu);
    const button = await findByRole('button');
    expect(button).toHaveTextContent('In English (EN)');
  });

  it('should match snapshot', async () => {
    const { baseElement, getByRole } = render(BasicLanguageMenu);
    const menuButton = getByRole('button') as HTMLButtonElement;
    expect(baseElement).toMatchSnapshot();
    await act(async () => {
      fireEvent.click(menuButton);
    });
    expect(baseElement).toMatchSnapshot();
  });
});

describe('movement in LanguageMenu', () => {
  const BasicLanguageMenu = TestLanguageMenu(languageMenuProps);

  it('should match snapshot', async () => {
    const { baseElement, getByRole } = render(BasicLanguageMenu);
    const menuButton = getByRole('button') as HTMLButtonElement;

    await act(async () => {
      fireEvent.click(menuButton);

      await new Promise((resolve) => {
        setTimeout(resolve, 100);
      });
    });

    await act(async () => {
      fireEvent.keyPress(baseElement, {
        key: 'ArrowDown',
      });

      await new Promise((resolve) => {
        setTimeout(resolve, 10);
      });
    });

    expect(baseElement).toMatchSnapshot();
  });
});

// Don't validate aria-attributes since Portal is not rendered and there is no pair for aria-controls
test(
  'should not have basic accessibility issues',
  axeTest(TestLanguageMenu(languageMenuProps), {
    rules: {
      'aria-valid-attr-value': {
        enabled: false,
      },
    },
  }),
);
