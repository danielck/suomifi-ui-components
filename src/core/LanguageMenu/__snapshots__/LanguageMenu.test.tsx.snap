// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic LanguageMenu should match snapshot 1`] = `
.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c2:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c2::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c2::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c2::-webkit-inner-spin-button {
  height: auto;
}

.c2::-webkit-outer-spin-button {
  height: auto;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c3 {
  vertical-align: baseline;
}

.c3.fi-icon {
  display: inline-block;
}

.c3 .fi-icon-base-fill {
  fill: currentColor;
}

.c3 .fi-icon-base-stroke {
  stroke: currentColor;
}

.c3.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c3.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c1.fi-language-menu {
  color: hsl(0,0%,13%);
}

.c1.fi-language-menu .fi-language-menu_button {
  color: hsl(0,0%,13%);
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  padding: 9px 10px;
  line-height: 24px;
  background-color: hsl(0,0%,100%);
  border: 1px solid transparent;
  border-radius: 2px;
  word-break: break-word;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_text_wrapper {
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  word-break: break-word;
  line-height: 24px;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_icon {
  width: 18px;
  height: 18px;
  margin-left: 10px;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_icon .fi-icon-base-fill {
  fill: hsl(212,63%,45%);
}

.c1.fi-language-menu .fi-language-menu_button:focus {
  outline: 0;
  position: relative;
}

.c1.fi-language-menu .fi-language-menu_button:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c1.fi-language-menu .fi-language-menu_button:hover {
  border-color: hsl(202,7%,80%);
}

.c1.fi-language-menu .fi-language-menu_button--open {
  border-color: hsl(202,7%,80%);
}

<body>
  <div>
    <div
      class="c0 fi-language-menu c1 lm-test"
      data-testid="language-menu-test-id"
      id="wrapper-id"
    >
      <button
        aria-controls="test-id-menu"
        aria-expanded="false"
        aria-haspopup="menu"
        class="c2 fi-language-menu_button"
        id="test-id"
        type="button"
      >
        In English (EN)
        <svg
          aria-hidden="true"
          class="fi-icon c3 fi-language-menu_button_icon"
          focusable="false"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="fi-icon-base-fill"
            d="M12.99 15.61c-.273.26-.632.39-.99.39s-.717-.13-.99-.39l-5.6-5.334a1.287 1.287 0 010-1.885 1.448 1.448 0 011.98 0l4.61 4.39 4.61-4.39a1.448 1.448 0 011.98 0c.547.52.547 1.365 0 1.885l-5.6 5.333z"
            fill="#222"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;

exports[`Basic LanguageMenu should match snapshot 2`] = `
.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c2:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c2::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c2::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c2::-webkit-inner-spin-button {
  height: auto;
}

.c2::-webkit-outer-spin-button {
  height: auto;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c5 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c5.fi-language-menu-popover {
  background-color: hsl(0,0%,100%);
  box-shadow: 0px 4px 8px 0px hsla(0,0%,13%,0.14);
  border: 1px solid hsl(0,0%,58%);
  border-radius: 2px;
  z-index: 888;
}

.c5.fi-action-menu-popover--hidden {
  visibility: hidden;
}

.c5 .fi-language-menu-popover_list {
  padding-top: 5px;
  padding-bottom: 5px;
  margin: 0;
  padding-left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  list-style-type: none;
  overflow-y: auto;
  max-height: 260px;
}

.c5 .fi-language-menu-popover_list:focus {
  outline: none;
}

.c5 .fi-language-menu-popover_popper-arrow::before,
.c5 .fi-language-menu-popover_popper-arrow::after {
  content: '';
  position: absolute;
  left: -9px;
  height: 0;
  width: 0;
  border: solid transparent;
  pointer-events: none;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='bottom-end'] {
  bottom: 100%;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='bottom-end']::before {
  border-bottom-color: hsl(0,0%,58%);
  border-width: 9px;
  margin-right: -9px;
  bottom: 100%;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='bottom-end']::after {
  border-bottom-color: hsl(0,0%,100%);
  border-width: 8px;
  margin-right: -9px;
  left: -8px;
  bottom: 100%;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='top-end'] {
  top: 100%;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='top-end']::before {
  border-top-color: hsl(0,0%,58%);
  border-width: 9px;
  margin-right: -9px;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='top-end']::after {
  border-top-color: hsl(0,0%,100%);
  border-width: 8px;
  margin-right: -9px;
  left: -8px;
}

.c3 {
  vertical-align: baseline;
}

.c3.fi-icon {
  display: inline-block;
}

.c3 .fi-icon-base-fill {
  fill: currentColor;
}

.c3 .fi-icon-base-stroke {
  stroke: currentColor;
}

.c3.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c3.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c1.fi-language-menu {
  color: hsl(0,0%,13%);
}

.c1.fi-language-menu .fi-language-menu_button {
  color: hsl(0,0%,13%);
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  padding: 9px 10px;
  line-height: 24px;
  background-color: hsl(0,0%,100%);
  border: 1px solid transparent;
  border-radius: 2px;
  word-break: break-word;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_text_wrapper {
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  word-break: break-word;
  line-height: 24px;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_icon {
  width: 18px;
  height: 18px;
  margin-left: 10px;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_icon .fi-icon-base-fill {
  fill: hsl(212,63%,45%);
}

.c1.fi-language-menu .fi-language-menu_button:focus {
  outline: 0;
  position: relative;
}

.c1.fi-language-menu .fi-language-menu_button:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c1.fi-language-menu .fi-language-menu_button:hover {
  border-color: hsl(202,7%,80%);
}

.c1.fi-language-menu .fi-language-menu_button--open {
  border-color: hsl(202,7%,80%);
}

.c6.fi-language-menu-item {
  color: hsl(0,0%,13%);
  margin: 5px 0;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  word-break: break-word;
  color: hsl(0,0%,13%);
  display: inline-block;
  width: 100%;
  padding: 0 20px 0 5px;
  border-left: 4px solid transparent;
}

.c6.fi-language-menu-item:active,
.c6.fi-language-menu-item:focus,
.c6.fi-language-menu-item:visited {
  outline: none;
  color: hsl(0,0%,13%);
}

.c6.fi-language-menu-item--isHighlighted {
  border-left-color: hsl(212,63%,45%);
  position: relative;
}

.c6.fi-language-menu-item--isHighlighted:after {
  content: '';
}

.c6.fi-language-menu-item--selected {
  font-weight: bold;
  border-left-color: hsl(212,63%,45%);
}

@media (forced-colors:active) {
  .c6.fi-language-menu-item--isHighlighted:after {
    position: absolute;
    left: 1px;
    right: 1px;
    top: 1px;
    bottom: 1px;
    border: solid 3px Highlight;
  }
}

<body>
  <div>
    <div
      class="c0 fi-language-menu c1 lm-test"
      data-testid="language-menu-test-id"
      id="wrapper-id"
    >
      <button
        aria-controls="test-id-menu"
        aria-expanded="true"
        aria-haspopup="menu"
        class="c2 fi-language-menu_button fi-language-menu_button--open"
        id="test-id"
        type="button"
      >
        In English (EN)
        <svg
          aria-hidden="true"
          class="fi-icon c3 fi-language-menu_button_icon"
          focusable="false"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="fi-icon-base-fill"
            d="M12.99 8.39A1.433 1.433 0 0012 8c-.358 0-.717.13-.99.39l-5.6 5.334a1.287 1.287 0 000 1.885 1.448 1.448 0 001.98 0l4.61-4.39 4.61 4.39a1.448 1.448 0 001.98 0 1.287 1.287 0 000-1.885l-5.6-5.333z"
            fill="#222"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        class="c4 c5 fi-language-menu-popover"
        role="menu"
        style="position: fixed; left: 0px; top: 0px; width: 0px; right: 0px; transform: translate(0px, 15px);"
      >
        <div
          aria-labelledby="test-id"
          class="c4 fi-language-menu-popover_list"
          id="test-id-menu"
          role="menu"
          tabindex="-1"
        >
          <button
            class="c2 fi-language-menu-item c6 fi-language-menu-item--isHighlighted"
            id="test-id-list-item-0"
            lang="fi"
            role="menuitem"
            tabindex="0"
            theme="[object Object]"
            type="button"
          >
            Suomeksi (FI)
          </button>
          <button
            class="c2 fi-language-menu-item c6"
            id="test-id-list-item-1"
            lang="sv"
            role="menuitem"
            tabindex="-1"
            theme="[object Object]"
            type="button"
          >
            På svenska (SV)
          </button>
          <button
            class="c2 fi-language-menu-item c6 fi-language-menu-item--selected"
            id="test-id-list-item-2"
            lang="en"
            role="menuitem"
            tabindex="-1"
            theme="[object Object]"
            type="button"
          >
            In English (EN)
          </button>
        </div>
        <div
          class="c0 fi-language-menu-popover_popper-arrow"
          data-popper-arrow="true"
          data-popper-placement="bottom-end"
          style="position: absolute; left: 0px; transform: translate(0px, 0px);"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`movement in LanguageMenu should match snapshot 1`] = `
.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c2:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c2::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c2::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c2::-webkit-inner-spin-button {
  height: auto;
}

.c2::-webkit-outer-spin-button {
  height: auto;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c5 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c5.fi-language-menu-popover {
  background-color: hsl(0,0%,100%);
  box-shadow: 0px 4px 8px 0px hsla(0,0%,13%,0.14);
  border: 1px solid hsl(0,0%,58%);
  border-radius: 2px;
  z-index: 888;
}

.c5.fi-action-menu-popover--hidden {
  visibility: hidden;
}

.c5 .fi-language-menu-popover_list {
  padding-top: 5px;
  padding-bottom: 5px;
  margin: 0;
  padding-left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  list-style-type: none;
  overflow-y: auto;
  max-height: 260px;
}

.c5 .fi-language-menu-popover_list:focus {
  outline: none;
}

.c5 .fi-language-menu-popover_popper-arrow::before,
.c5 .fi-language-menu-popover_popper-arrow::after {
  content: '';
  position: absolute;
  left: -9px;
  height: 0;
  width: 0;
  border: solid transparent;
  pointer-events: none;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='bottom-end'] {
  bottom: 100%;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='bottom-end']::before {
  border-bottom-color: hsl(0,0%,58%);
  border-width: 9px;
  margin-right: -9px;
  bottom: 100%;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='bottom-end']::after {
  border-bottom-color: hsl(0,0%,100%);
  border-width: 8px;
  margin-right: -9px;
  left: -8px;
  bottom: 100%;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='top-end'] {
  top: 100%;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='top-end']::before {
  border-top-color: hsl(0,0%,58%);
  border-width: 9px;
  margin-right: -9px;
}

.c5 .fi-language-menu-popover_popper-arrow[data-popper-placement^='top-end']::after {
  border-top-color: hsl(0,0%,100%);
  border-width: 8px;
  margin-right: -9px;
  left: -8px;
}

.c3 {
  vertical-align: baseline;
}

.c3.fi-icon {
  display: inline-block;
}

.c3 .fi-icon-base-fill {
  fill: currentColor;
}

.c3 .fi-icon-base-stroke {
  stroke: currentColor;
}

.c3.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c3.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c1.fi-language-menu {
  color: hsl(0,0%,13%);
}

.c1.fi-language-menu .fi-language-menu_button {
  color: hsl(0,0%,13%);
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  padding: 9px 10px;
  line-height: 24px;
  background-color: hsl(0,0%,100%);
  border: 1px solid transparent;
  border-radius: 2px;
  word-break: break-word;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_text_wrapper {
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  word-break: break-word;
  line-height: 24px;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_icon {
  width: 18px;
  height: 18px;
  margin-left: 10px;
}

.c1.fi-language-menu .fi-language-menu_button .fi-language-menu_button_icon .fi-icon-base-fill {
  fill: hsl(212,63%,45%);
}

.c1.fi-language-menu .fi-language-menu_button:focus {
  outline: 0;
  position: relative;
}

.c1.fi-language-menu .fi-language-menu_button:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c1.fi-language-menu .fi-language-menu_button:hover {
  border-color: hsl(202,7%,80%);
}

.c1.fi-language-menu .fi-language-menu_button--open {
  border-color: hsl(202,7%,80%);
}

.c6.fi-language-menu-item {
  color: hsl(0,0%,13%);
  margin: 5px 0;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  word-break: break-word;
  color: hsl(0,0%,13%);
  display: inline-block;
  width: 100%;
  padding: 0 20px 0 5px;
  border-left: 4px solid transparent;
}

.c6.fi-language-menu-item:active,
.c6.fi-language-menu-item:focus,
.c6.fi-language-menu-item:visited {
  outline: none;
  color: hsl(0,0%,13%);
}

.c6.fi-language-menu-item--isHighlighted {
  border-left-color: hsl(212,63%,45%);
  position: relative;
}

.c6.fi-language-menu-item--isHighlighted:after {
  content: '';
}

.c6.fi-language-menu-item--selected {
  font-weight: bold;
  border-left-color: hsl(212,63%,45%);
}

@media (forced-colors:active) {
  .c6.fi-language-menu-item--isHighlighted:after {
    position: absolute;
    left: 1px;
    right: 1px;
    top: 1px;
    bottom: 1px;
    border: solid 3px Highlight;
  }
}

<body>
  <div>
    <div
      class="c0 fi-language-menu c1 lm-test"
      data-testid="language-menu-test-id"
      id="wrapper-id"
    >
      <button
        aria-controls="test-id-menu"
        aria-expanded="true"
        aria-haspopup="menu"
        class="c2 fi-language-menu_button fi-language-menu_button--open"
        id="test-id"
        type="button"
      >
        In English (EN)
        <svg
          aria-hidden="true"
          class="fi-icon c3 fi-language-menu_button_icon"
          focusable="false"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="fi-icon-base-fill"
            d="M12.99 8.39A1.433 1.433 0 0012 8c-.358 0-.717.13-.99.39l-5.6 5.334a1.287 1.287 0 000 1.885 1.448 1.448 0 001.98 0l4.61-4.39 4.61 4.39a1.448 1.448 0 001.98 0 1.287 1.287 0 000-1.885l-5.6-5.333z"
            fill="#222"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        class="c4 c5 fi-language-menu-popover"
        role="menu"
        style="position: fixed; left: 0px; top: 0px; width: 0px; right: 0px; transform: translate(0px, 15px);"
      >
        <div
          aria-labelledby="test-id"
          class="c4 fi-language-menu-popover_list"
          id="test-id-menu"
          role="menu"
          tabindex="-1"
        >
          <button
            class="c2 fi-language-menu-item c6 fi-language-menu-item--isHighlighted"
            id="test-id-list-item-0"
            lang="fi"
            role="menuitem"
            tabindex="0"
            theme="[object Object]"
            type="button"
          >
            Suomeksi (FI)
          </button>
          <button
            class="c2 fi-language-menu-item c6"
            id="test-id-list-item-1"
            lang="sv"
            role="menuitem"
            tabindex="-1"
            theme="[object Object]"
            type="button"
          >
            På svenska (SV)
          </button>
          <button
            class="c2 fi-language-menu-item c6 fi-language-menu-item--selected"
            id="test-id-list-item-2"
            lang="en"
            role="menuitem"
            tabindex="-1"
            theme="[object Object]"
            type="button"
          >
            In English (EN)
          </button>
        </div>
        <div
          class="c0 fi-language-menu-popover_popper-arrow"
          data-popper-arrow="true"
          data-popper-placement="bottom-end"
          style="position: absolute; left: 0px; transform: translate(0px, 0px);"
        />
      </div>
    </div>
  </div>
</body>
`;
